numba_rvsdg/tests/test_ast_transforms.py

```diff
@@ -31,9 +31,9 @@ def compare(
         self,
         function: Callable[..., Any],
         expected: dict[str, dict[str, Any]],
-        unreachable: set[int] = set(),
-        empty: set[int] = set(),
-        arguments: list[any] = [],
+        unreachable: set[str] = set(),
+        empty: set[str] = set(),
+        arguments: list[Any] = [],
```

Comment:
This might be unrelated but I am not sure it's good idea to have a list as a default argument? (Since it is mutable, it's generally not recommended to do that? )

Reply:
In principle, yes. In this case the compare method treats these mutable structures as immutable, and so it's ok to define them as empty here. They are either empty or contain something, but will never be modified. It's a bit shorter than handing in `None` and then checking at the start of the function if the argument is `None` and the instantiating an empty container as a result.